src/user_input.rs

```diff
@@ -1,444 +1,814 @@
+    /// An optional [`MouseButton`] [`Input`] stream
+    pub mouse: Option<&'a Input<MouseButton>>,
+    /// The [`Gamepad`] that this struct will detect inputs from
+    pub associated_gamepad: Option<Gamepad>,
+}
+
+// Constructors
+impl<'a> InputStreams<'a> {
+    /// Construct [`InputStreams`] with only a [`GamepadButton`] input stream
+    pub fn from_gamepad(
+        gamepad_button_stream: &'a Input<GamepadButton>,
+        gamepad_button_axis_stream: &'a Axis<GamepadButton>,
+        gamepad_axis_stream: &'a Axis<GamepadAxis>,
+        associated_gamepad: Gamepad,
+    ) -> Self {
+        Self {
+            gamepad_buttons: Some(gamepad_button_stream),
+            gamepad_button_axes: Some(gamepad_button_axis_stream),
+            gamepad_axes: Some(gamepad_axis_stream),
+            keyboard: None,
+            mouse: None,
+            associated_gamepad: Some(associated_gamepad),
+        }
+    }
+
+    /// Construct [`InputStreams`] with only a [`KeyCode`] input stream
+    pub fn from_keyboard(keyboard_input_stream: &'a Input<KeyCode>) -> Self {
+        Self {
+            gamepad_buttons: None,
+            gamepad_button_axes: None,
+            gamepad_axes: None,
+            keyboard: Some(keyboard_input_stream),
+            mouse: None,
+            associated_gamepad: None,
+        }
+    }
+
+    /// Construct [`InputStreams`] with only a [`GamepadButton`] input stream
+    pub fn from_mouse(mouse_input_stream: &'a Input<MouseButton>) -> Self {
+        Self {
+            gamepad_buttons: None,
+            gamepad_button_axes: None,
+            gamepad_axes: None,
+            keyboard: None,
+            mouse: Some(mouse_input_stream),
+            associated_gamepad: None,
+        }
+    }
+}
+
+// Input checking
+impl<'a> InputStreams<'a> {
+    /// Is the `input` matched by the [`InputStreams`]?
+    pub fn input_pressed(&self, input: &UserInput) -> bool {
+        match input {
+            UserInput::Single(button) => self.button_pressed(*button),
+            UserInput::Chord(buttons) => self.all_buttons_pressed(buttons),
+            UserInput::VirtualDPad {
+                up,
+                down,
+                left,
+                right,
+            } => {
+                for button in [up, down, left, right] {
+                    if self.button_pressed(*button) {
+                        return true;
+                    }
+                }
+                false
+            }
+        }
+    }
+
+    /// Is at least one of the `inputs` pressed?
+    #[must_use]
+    pub fn any_pressed(&self, inputs: &PetitSet<UserInput, 16>) -> bool {
+        for input in inputs.iter() {
+            if self.input_pressed(input) {
+                return true;
+            }
+        }
+        // If none of the inputs matched, return false
+        false
+    }
+
+    /// Is the `button` pressed?
+    #[must_use]
+    pub fn button_pressed(&self, button: InputKind) -> bool {
```

Comment:
@alice-i-cecile `InputKind` now includes axes. Should this method be renamed (someday)? 🤷🏻‍♂️ 

```suggestion
    /// Is the `button` pressed?
    /// TODO: Rename this method since InputKind includes axes as well as buttons?
    #[must_use]
    pub fn button_pressed(&self, button: InputKind) -> bool {
```

Reply:
Agreed; see #157.